letsencrypt/auth_handler.py

```diff
@@ -153,13 +153,18 @@ def _send_responses(self, achalls, resps, chall_update):
         """
         active_achalls = []
         for achall, resp in itertools.izip(achalls, resps):
+            # XXX: make sure that all achalls, including those
+            # corresponding to None or False returned from
+            # Authenticator are removed from the queue and thus avoid
+            # infinite loop
+            active_achalls.append(achall)
```

Comment:
Hmm... this shouldn't be moved out.  If it is pulled out... it tells the authenticator to clean up challenges that haven't been solved yet.  At least that wasn't the intended API...